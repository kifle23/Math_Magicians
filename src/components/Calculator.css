.calculator {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: auto;
  width: 40%;
  min-width: 500px;
  background-color: #fff;
  box-shadow: 0 0 20px #666;
  border-radius: 2px;
  position: absolute;
  top: 20vh;
  left: 30vw;
}

.display {
  height: 80px;
  background-color: #858694;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  font-size: 24px;
  padding: 0 10px;
  color: #fff;
}

.buttons {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  min-height: 300px;
}

button {
  padding: 1.5rem;
  cursor: pointer;
  background-color: #e0e0e0;
  border-color: #80646479;
  border-radius: 0;
  border-style: solid;
}

.button-0 {
  grid-column: span 2;
}

.operator {
  background-color: #f5913e;
}
