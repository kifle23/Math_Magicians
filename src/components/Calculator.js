import React from 'react';
import PropTypes from 'prop-types';
import './Calculator.css';

function Display({ value }) {
  return <div className="display">{value}</div>;
}

function Calculator() {
  return (
    <div className="calculator">
      <Display value={0} />
      <div className="buttons">
        <button type="button">AC</button>
        <button type="button">+/-</button>
        <button type="button">% </button>
        <button type="button" className="operator">
          รท
        </button>
        <button type="button">7</button>
        <button type="button">8</button>
        <button type="button">9</button>
        <button type="button" className="operator">
          x
        </button>
        <button type="button">4</button>
        <button type="button">5</button>
        <button type="button">6</button>
        <button type="button" className="operator">
          -
        </button>
        <button type="button">1</button>
        <button type="button">2</button>
        <button type="button">3</button>
        <button type="button" className="operator">
          +
        </button>
        <button type="button" className="button-0">
          0
        </button>
        <button type="button">.</button>
        <button type="button" className="operator">
          =
        </button>
      </div>
    </div>
  );
}

Display.propTypes = {
  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
};

export default Calculator;
