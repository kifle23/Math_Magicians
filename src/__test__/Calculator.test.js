import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator', () => {
  test('renders correctly', () => {
    const { container } = render(<Calculator />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('updates display when buttons are clicked', () => {
    const { container, getByText } = render(<Calculator />);
    fireEvent.click(getByText('1'));
    fireEvent.click(getByText('+'));
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('='));
    expect(container.firstChild).toMatchSnapshot();
  });
});
