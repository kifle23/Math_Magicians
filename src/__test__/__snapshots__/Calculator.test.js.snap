// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<div
  class="some-math"
>
  Let's do some math!
</div>
`;

exports[`Calculator updates display when buttons are clicked 1`] = `
<div
  class="some-math"
>
  Let's do some math!
</div>
`;
